lib/server/services/run/RunService.js

```diff
@@ -54,7 +55,11 @@ class RunService {
      * @return {Promise<Run>} resolve with the created run instance
      */
     async create(newRun) {
-        const runId = await createRun(runAdapter.toDatabase(newRun));
+        // Save the old detector values
+        newRun.concatenatedDetectors = newRun.detectors;
```

Comment:
I think here it would be easier to use sequelize hooks on create and update:

https://sequelize.org/docs/v6/other-topics/hooks/